x/evm/statedb/statedb.go

```diff
@@ -612,11 +612,14 @@ func (s *StateDB) Commit() error {
 				return errorsmod.Wrap(err, "failed to delete account")
 			}
 		} else {
-			if obj.code != nil && obj.dirtyCode {
+			codeDirty := obj.codeDirty()
+			if codeDirty && obj.code != nil {
 				s.keeper.SetCode(s.ctx, obj.CodeHash(), obj.code)
 			}
-			if err := s.keeper.SetAccount(s.ctx, obj.Address(), obj.account); err != nil {
-				return errorsmod.Wrap(err, "failed to set account")
+			if codeDirty || obj.nonceDirty() {
```

Comment:
do we miss check empty here

Reply:
what empty?

Reply:
what do you mean? delete empty accounts?

Reply:
I mean we also need SetAccount for new account?

Reply:
i see, otherwise,it'll be consensus breaking.

Reply:
done